### PR TITLE
feat: initial access tags work

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Add the ability to attach access management tags to resources created by the module.

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Add ability to attach access tags to resources created by the module, consumers who want to use the access tags will need additional permissions for Tagging service at the Administrator level

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by MatthewLemmond** [comment]
@toddgiguere @ocofaigh this PR is failing upgrade test due to passing in the access tags variable in the test which was previously not used in the module, should this test be skipped/run manually for verification?

**Comment by MatthewLemmond** [comment]
I have manually simulated an upgrade test and found that no resources are created or destroyed if access tags are not set, and also that setting the tags then provides the following output
```
Terraform will perform the following actions:

  # module.ocp_base.ibm_resource_tag.cluster_access_tag[0] will be created
  + resource "ibm_resource_tag" "cluster_access_tag" {
      + acccount_id = (known after apply)
      + id          = (known after apply)
      + resource_id = "crn:v1:bluemix:public:containers-kubernetes:eu-gb:a/abac0df06b644a9cabc6e44f55b3880e:cgmrkn2l06jhbv8vbccg::"
      + tag_type    = "access"
      + tags        = [
          + "geretain-dev:permanent-test-tag-1",
          + "geretain-dev:permanent-test-tag-2",
        ]
    }

  # module.ocp_base.ibm_resource_tag.cos_access_tag[0] will be created
  + resource "ibm_resource_tag" "cos_access_tag" {
      + acccount_id = (known after apply)
      + id          = (known after apply)
      + resource_id = "crn:v1:bluemix:public:cloud-object-storage:global:a/abac0df06b644a9cabc6e44f55b3880e:0427c15f-1bcb-41c1-b0d0-bf52ad34472d::"
      + tag_type    = "access"
      + tags        = [
          + "geretain-dev:permanent-test-tag-1",
          + "geretain-dev:permanent-test-tag-2",
        ]
    }

Plan: 2 to add, 0 to change, 0 to destroy.
```

Given that this is a non-destructive change, and that only the newly added resource tags are included in the plan, I am going to skip the upgrade test for this PR.

**Comment by MatthewLemmond** [comment]
I have manually run the upgrade test and have found that not having the access tags set results in no changes after the main branch apply, and setting the access tags variable provides the following output:
```
Terraform will perform the following actions:

  # module.ocp_base.ibm_resource_tag.cluster_access_tag[0] will be created
  + resource "ibm_resource_tag" "cluster_access_tag" {
      + acccount_id = (known after apply)
      + id          = (known after apply)
      + resource_id = "<CRN>"
      + tag_type    = "access"
      + tags        = [
          + "test-tag:tag-1",
          + "test-tag:tag-2",
        ]
    }

  # module.ocp_base.ibm_resource_tag.cos_access_tag[0] will be created
  + resource "ibm_resource_tag" "cos_access_tag" {
      + acccount_id = (known after apply)
      + id          = (known after apply)
      + resource_id = "<CRN>"
      + tag_type    = "access"
      + tags        = [
          + "test-tag:tag-1",
          + "test-tag:tag-2",
        ]
    }

Plan: 2 to add, 0 to change, 0 to destroy.
```

Given that this is a non-destructive change, I am going to skip the upgrade test as the only changes when running it are only the resource tags added in this PR.

Note: the only reason the upgrade test was failing is because we are passing in the access_tags variable to the test, but this is a new variable so terraform was erroring due to the unknown variable being provided to the main-branch version of the run.

**Review by toddgiguere — DISMISSED** [review]

**Review by ocofaigh — COMMENTED** [review]
Do you think there is a use case where someone may want different access tags for the different resources created? In this case COS and OCP cluster? If so, then we would need one variable per resource. Thoughts @MatthewLemmond @vburckhardt @toddgiguere ?

**Comment by MatthewLemmond** [comment]
@ocofaigh since we're adding these tags to other modules, including COS, that use-case would be covered by using an existing COS instance in the module. I think if we add a separate access_tags list for each optional resource created in a module it may become confusing having several different access tag variables for every different type of resource created

**Comment by ocofaigh** [comment]
@MatthewLemmond what about the `ibm_container_vpc_worker_pool` resource?

**Comment by MatthewLemmond** [comment]
@ocofaigh The access tag is associated with a resource via the resource's CRN and the worker pools themselves do not have any CRNs so they cannot have access tags attached to them.

From my testing with other scenarios like Key Protect keys, even though they have a CRN and are tagged with access management tags, the visibility of the keys wasn't affected by tagging them.

**Comment by ocofaigh** [comment]
Thanks @MatthewLemmond 
Can we check with the wider team about using the value of `var.access_tags` for both cluster and cos? I think I would still like them as separate variables, but maybe we run it by the team for feedback?

**Review by shemau — DISMISSED** [review]
Looks good

File: tests/pr_test.go [hunk 1]
```diff
@@ -2,30 +2,55 @@
...
 var sharedInfoSvc *cloudinfo.CloudInfoService
+var existingAccessTags []string
 
 // TestMain will be run before any parallel tests, used to set up a shared InfoService object to track region usage
 // for multiple tests
 func TestMain(m *testing.M) {
-
 	sharedInfoSvc, _ = cloudinfo.NewCloudInfoServiceFromEnv("TF_VAR_ibmcloud_api_key", cloudinfo.CloudInfoServiceOptions{})
+
+	// Read the YAML file contents
```
Comment (ocofaigh) [inline tests/pr_test.go:39]: There is now an inbuilt function to do this called `LoadMapFromYaml` (see https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/blob/main/tests/pr_test.go#L141). It makes the code easier if you wanted to roll it out here?

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**